src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -216,14 +216,10 @@ export const useDiscoverHistogram = ({
   /**
    * Request params
    */
-  const { query, filters } = useQuerySubscriber({ data: services.data });
+  const { filters } = useQuerySubscriber({ data: services.data });
   const customFilters = useInternalStateSelector((state) => state.customFilters);
-  const timefilter = services.data.query.timefilter.timefilter;
-  const timeRange = timefilter.getAbsoluteTime();
-  const relativeTimeRange = useObservable(
-    timefilter.getTimeUpdate$().pipe(map(() => timefilter.getTime())),
-    timefilter.getTime()
-  );
+  const query = main$.getValue().query;
+  const { timeRangeRelative, timeRange } = useObservable(main$, main$.getValue());
```

Comment:
@davismcphee this could be 

```suggestion
  const { query, timeRangeRelative, timeRange } = useObservable(main$, main$.getValue());
```

Not 100% but there might have been a race between query and timerange before, when loading the saved search. With this change it's just one source of truth, the main$, which is updated when a data fetching is triggered. we could do the same filters also customFilters. One source of truth ... it could also be the internal state container holding the truth, just not many different sources.